### PR TITLE
Updates with the drop images list and symlinks

### PR DESCRIPTION
**Pull request by San411**
I have just committed the changes i think are needed. 
I lost some images while removing the improper symlinks but by using os.symlink() method i could move forward with the code.
Better to just include the images that are common in both the csv file and downloaded images directory. Changed this in both download_images and train_cnn files.
In the readme file i have just added the symlink verification.

Now you can do changes in just a single commit.
Thanks.

### CODE REVIEW

**Comment by jmathur25** [comment]
I'm still not sure that the code to drop un-downloaded images needs to change

**Comment by San411** [comment]
I assume you're referring to the intersection() method is used to filter the 'idx_not_download' list.
There were some images downloaded in the respective country images directory which were not present in the "image_download_locs.csv" file due to which, in the following code

idx_not_download = df_download.set_index('image_name').drop(downloaded)['row'].values.tolist()
df_download.drop(idx_not_download, inplace=True)

the variable 'downloaded' has some image names that is not present in the 'image_name' column and the line showed an error. By including the image names that are common in both the lists it is easier to use drop() method and will still give the same result.

However if you still think is it not necessary there is no need to make that change

File: README.md [hunk 1]
```diff
@@ -45,7 +55,7 @@ conda install -c conda-forge ipywidgets
...
+3) get an api key from either Planet or Google's Static Maps API service. Both of these should be free, but Planet may take some time to approve and require you to list a research project to be eligible for the free tier. Google's service should be free if you download under 100k per month.To enable access to google API the verification process has to be completed by providing your card details(no charges will be applied!). Save the api keys to `planet_api_key.txt` or `google_api_key.txt` in the root directory. I used Planet's API because then I could download images from 2015 and 2016, whereas Google's service only offers recent images over the last year. The code will show how to get the images from the API.
```
Comment (jmathur25) [inline README.md]: remove this change

File: scripts/train_cnn.ipynb [hunk 2]
```diff
@@ -211,8 +211,11 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "idx_not_download = df_download.set_index('image_name').drop(downloaded)['row'].values.tolist()\n",
-    "df_download.drop(idx_not_download, inplace=True)"
+    "downloaded =  list(set(downloaded).intersection(set(df_download['image_name'])))\n",
```
Comment (jmathur25) [inline scripts/train_cnn.ipynb]: why not do `df_download = df_download.set_index('image_name')[downloaded].reset_index()`

File: README.md [hunk 3]
```diff
@@ -27,7 +27,17 @@ conda activate <ENV_NAME>
...
+The libraries that are most likely to fail are gdal and geoio. If a requirement fails to install, first make sure you follow this install procedure exactly. Using `pip` to install GDAL did not work for me, and the only way I got it to install was by including it when I first make the conda environment (hence `pip gdal`). There are also several Stack Overflow posts on these issues, so hopefully one will work on your machine. Regarding machines, I'd highly recommend getting a deep learning VM or some other CUDA-enabled runtime environment. The operating system should preferably be Linux. If not, you may run into some issues creating the symlinks. If the symlinks still don't work then I would recommend to simply remove the broken symlinks.
+
+Follow this to remove the broken symlinks:
```
Comment (jmathur25) [inline README.md]: I don't think this needs to be included in the readme.

**Pull request merged**